packages/bitcore-client/src/wallet.ts

```diff
@@ -82,14 +83,20 @@ export class Wallet {
   }
 
   static async create(params: Partial<WalletObj>) {
-    const { chain, network, name, phrase, password, path, lite, baseUrl } = params;
+    const { chain, network, name, phrase, xpriv, password, path, lite, baseUrl } = params;
     let { storageType, storage } = params;
     if (!chain || !network || !name) {
       throw new Error('Missing required parameter');
     }
     // Generate wallet private keys
-    const mnemonic = new Mnemonic(phrase);
-    const hdPrivKey = mnemonic.toHDPrivateKey('', network).derive(Deriver.pathFor(chain, network));
+    let hdPrivKey;
+    let mnemonic;
+    if (xpriv) {
```

Comment:
Just thinking about edge cases - we should check for mutual exclusivity between the phrase and xpriv params

Reply:
Good call. Added in latest commit.